android/quest/src/main/java/org/smartregister/fhircore/quest/ui/report/measure/MeasureReportViewModel.kt

```diff
@@ -360,6 +389,27 @@ constructor(
       }
   }
 
+  /** Returns a Map of year-month list for for all months falling in given measure period */
+  fun getReportGenerationRange(
+    startDate: Date? =
+      measureReportRepository.getCampaignStartDate().getYyyMmDd(MEASURE_REPORT_DATE_FORMAT)
+  ): Map<String, List<ReportRangeSelectionData>> {
+
+    val yearMonths = mutableListOf<ReportRangeSelectionData>()
+    val endDate = Calendar.getInstance().time.asYyyyMmDd().getYyyMmDd(MEASURE_REPORT_DATE_FORMAT)
+    var lastDate = endDate?.firstDayOfMonth()
+
+    while (lastDate!!.after(startDate)) {
+      yearMonths.add(ReportRangeSelectionData(lastDate.asMmmm(), lastDate.asYyyy(), lastDate))
+
+      lastDate = lastDate.plusMonths(-1)
+    }
+    return yearMonths.toList().groupBy { it.year }
+  }
+
+  /** * show Fixed month/year listing */
```

Comment:
can we make this a full docstring as well, I know we haven't consistent in the past but we want to try and do better going forward :)